### PR TITLE
RavenDB-8791 Use `eval` with function `this` to parse the parameters

### PR DESCRIPTION
**Pull request by fitzchak**

### CODE REVIEW

File: src/Raven.Studio/wwwroot/Content/ace/mode-rql.js [hunk 1]
```diff
@@ -1037,3 +1037,7 @@ oop.inherits(Mode, TextMode);
 exports.Mode = Mode;
 
 });
+
+this.window.evalUnstrict = function (str) {
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/Content/ace/mode-rql.js]: Can we put this in some *.ts file?

**Review by ml054 — CHANGES_REQUESTED** [review]
Can you use new Function('here goes function body') instead? eval overrides global objects + it is not secure.

**Comment by ml054** [comment]
Optionally you can use es6 proxies to intercept global variables creation (or to catching assigning to this) + 'with' keyword to sandbox execution.

File: src/Raven.Studio/wwwroot/Content/ace/mode-rql.js (hunk 1 above)
Comment (fitzchak) [reply src/Raven.Studio/wwwroot/Content/ace/mode-rql.js]: This was removed. I had to put it there to be outside the 'use strict' mode, which isn't possible in .ts file.

**Pull request merged**